src/containers/rosterChecker/rosterChecker.jsx

```diff
@@ -0,0 +1,258 @@
+const PRESENT = "present";
+const MISSING = "missing";
+const SHAMROCK_GREEN = "#0F9D58";
+const CINNABAR_RED = "#EA4335";
+
+const styles = theme => ({
+    sheet: {
+        marginLeft: 10,
+        marginTop: 10,
+        marginBottom: 10,
+        marginRight: 10,
+        borderStyle: "solid",
+        borderWidth: "1px",
+        borderColor: "#777"
+    },
+    card: {
+        background:"#fafafa"
+    },
+    cardHeader: {
+        paddingLeft: 10,
+        paddingTop: 10,
+        paddingBottom: 0,
+        paddingRight: 10
+    },
+    cardContent: {
+        height: "calc(100vh - 250px)",
+        overflow: "scroll"
+    }
+})
+
+class SheetView extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            arrangementId: "",
+            currentRoster: [],
+            newRoster: [],
+        }
+    }
+
+    generateColumnHeaders = () => {
+        return this.generateColumnDefs().map(col => col.data);
+    }
+
+    generateColumnDefs = () => {
+        return [
+            {
+                data: NAME_FIELD,
+                width: 220
+            },
+            {  
+                data: VALIDITY_FIELD,
+                width: 220,
+                readOnly: true,
+                renderer: this.renderValidity
+            }
+        ];
+    }
+
+    generateCurrentRoster = () => {
+        const itemList = []
+        console.log(this.props.real)
+        if (!isEmpty(this.props.real)) {
+            console.log("generating item list...");
+        
+            this.props.real.items.forEach(item => {
+                const validityOfItem = this.checkIfNameInRoster(item.name, this.state.newRoster) ? PRESENT : MISSING;
+                itemList.push({name: item.name, validity: validityOfItem});
+            });
+            this.setState({...this.state, currentRoster: itemList});
+        }
+        return itemList;
+    }
+
+    getItemFromItemName = (itemName) => {
+        return this.props.real.items.find(x => x.name === itemName);
+    }
+
+    renderValidity = (instance, td, row, col, prop, value, cellProperties) => {
+        if (value === PRESENT) {
+            td.style.backgroundColor = SHAMROCK_GREEN;
+        }
+        else if (value === MISSING) {
+            td.style.backgroundColor = CINNABAR_RED;
+        }
+        return td.innerHTML = value;
+    }
+
+    checkIfNameInRoster = (name, roster) => {
+        return roster.filter(row => {
+            return row.name === name}).length > 0;
+    }
+
+    processChangeItemName = (row, columnTitle, previous, current) => {
+        const newRosterClone = cloneDeep(this.state.newRoster);
+        const validityOfItem = this.checkIfNameInRoster(current, this.state.currentRoster) ? PRESENT : MISSING;
+        newRosterClone[row] = {name: current, validity: validityOfItem}
+        this.setState({...this.state, newRoster: newRosterClone})
+    }
+
+    processChange = (row, columnTitle, previous, current) => {
```

Comment:
what is prev/current?

Reply:
previous is the previous entry of the cell, current is the current entry of the cell.